packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -159,14 +160,47 @@ export default class Deploy extends CLICommand {
     this.validateDeployConfigJson(deployConfigPath, deployConfig);
     const hostedPage = this.buildHostedPage(deployConfig);
 
-    if (flags.pageId) {
-      await this.updatePage({...hostedPage, id: flags.pageId});
+    const pageId = flags.pageId || (await this.getPageIdFromPage(hostedPage));
+
+    if (pageId) {
+      await this.updatePage({...hostedPage, id: pageId});
       return;
     }
 
     await this.createPage(hostedPage);
   }
 
+  private async getPageIdFromPage(
+    hostedPage: New<HostedPage, null>
+  ): Promise<string | null> {
+    const {flags} = await this.parse(Deploy);
+    const orgId = getTargetOrg(this.configuration, flags.organization);
+    const client = await this.createClient();
+    const response = await client.hostedPages.list({
+      filter: hostedPage.name,
+      perPage: 1000,
+    });
+    const page = response.items
+      .filter((page) => page.name === hostedPage.name)
+      .pop();
+
+    if (!page) {
+      return null;
+    }
+    if (await this.confirmOverwrite(page.id, orgId)) {
+      return page.id;
+    } else {
+      this.error(
+        `Page name must be unique, try changing the "name" field in "coveo.deploy.json".`
```

Comment:
How is the id related to name?

Reply:
Essentially:
- there's a bijection between `organizationId+pageName` and the `pageId`
- pageIds are unique.
- Therefore, for a given organization, a page name must be unique in this organization